### PR TITLE
speed up build times

### PR DESCRIPTION
**Pull request by Toffikk**

### CODE REVIEW

**Comment by Titaniumtown** [comment]
How does this speed up build times?

**Comment by PaulBGD** [comment]
Maybe the incremental part? I'm not familiar with it though.

**Comment by jpenilla** [comment]
Configuring UTF-8 encoding for Javadoc and ProcessResources is not a bad idea, however this PR does absolutely nothing whatsoever in terms of improving build times. In fact, this PR will probably slow down build times by forcing a fork and applying the java plugin to the root project where before it was only applied to subprojects.

File: build.gradle.kts [hunk 1]
```diff
@@ -19,15 +19,16 @@ dependencies {
...
+allprojects {
     apply(plugin = "java")
 
     java { toolchain { languageVersion.set(JavaLanguageVersion.of(16)) } }
 
-    tasks.withType<JavaCompile>().configureEach {
-        options.encoding = "UTF-8"
-        options.release.set(16)
-    }
+    tasks.withType<JavaCompile> { options.isFork = true; options.isIncremental = true; options.encoding = Charsets.UTF_8.name(); options.release.set(16) }
```
Comment (jpenilla) [inline build.gradle.kts]: Have you heard of this thing called a newline?

**Review by jpenilla — CHANGES_REQUESTED** [review]

**Comment by jpenilla** [comment]
> Maybe the incremental part? I'm not familiar with it though.

It defaults to true. There is no need to set it manually.

**Comment by Toffikk** [comment]
> Configuring UTF-8 encoding for Javadoc and ProcessResources is not a bad idea, however this PR does absolutely nothing whatsoever in terms of improving build times. In fact, this PR will probably slow down build times by forcing a fork and applying the java plugin to the root project where before it was only applied to subprojects.

The Java plugin was applied to the root project before too, and forcing a fork can speed up the build times in some cases

**Comment by jpenilla** [comment]
I hadn't noticed it was applied to the root project, although it shouldn't be necessary to apply there, as the root project does not compile any java sources. It's possible paperweight is using it for something, I don't remember off the top of my head.

Regardless, it doesn't answer the question of why you are changing that block to apply to allprojects instead of subprojects. Those repositories and the java plugin don't need to be configured for the root project as it doesn't compile any java sources or pull anything from those repositories. I assume *maybe* it's because you want your changes to the java plugin to apply to the root project, but like I mentioned already it doesn't compile any java sources.

I also highly doubt that forcing a new process to be forked regardless of whether it's necessary will speed things up, feel free to provide a benchmark, or elaborate on what you mean by "in some cases" though.

**Comment by PaulBGD** [comment]
Unconvinced this improves things, please benchmark it from a clean build.

**Comment by PaulBGD** [comment]
Closing due to lack of a response.

**Pull request closed without merging**